lib/STM.ml

```diff
@@ -171,7 +171,35 @@ struct
     let res_arr = Array.map (fun c -> Domain.cpu_relax(); Spec.run c sut) cs_arr in
     List.combine cs (Array.to_list res_arr)
 
-  let rec check_obs pref cs1 cs2 s = match pref with
+  (* checks that all interleavings of a cmd triple satisfies all preconditions *)
+  let rec all_interleavings_ok pref cs1 cs2 s =
+    match pref with
```

Comment:
As the prefix is sequential, it should be valid by construction (?). So, do we really need to check it here or could we just compute the `spawn_state`?

Reply:
Good point! :+1: 
From a generation perspective I completely agree.
From a shrinking perspective, I don't think we can be sure that the prefix continues to be valid `precond`-wise, e.g., if the `cmd list` shrinker removes `cmd`s from the sequential prefix...

Such shrinking reminds me that it would be good to actually test that this combination works - a negative test (triggering shrinking) with a non-constant `precond`....